### ISSUE TITLE
Ability to capture LSan messages as ASan has

### ISSUE THREAD

**Issue by diane2332** [google/sanitizers#1121]
Please see gcc bug:
 91203 - __asan_set_error_report_callback has no effect on leak messages

I'm using ASan on a very large program where stderr is redirected to /dev/null, and I cannot change that. We are already capturing ASan messages by setting up a callback function, but that does not capture leak messages.

The best thing from my point of view would be for LSan to call the ASan callback function for messages. But if this doesn't make sense for running LSan independently, please create:

  void __lsan_set_error_report_callback (void (*callback)(const char*));

And for sure this lack of capturing LSan messages is unexpected and so must be well-documented where __asan_set_error_report_callback is documented, although I don't think it's documented officially anyway. It should be. I think I had to go through the code to find it.

**Comment by kcc** [comment]
Did you try
  * __sanitizer_set_death_callback
  * ASAN_OPTIONS=log_path=/tmp/zzz
?

**Comment by diane2332** [comment]
Yes we are using LSAN_OPTIONS=log_path=...  as a workaround. However, it's not a good one for several reasons having to do with the configuration of the application that I cannot put here.

I don't think that __sanitizer_set_death_callback will help with emitting messages. The callback function takes no arguments. Given the name, I assume it's only called when a severe error happens, not if there is a leak.

**Comment by ecatmur** [comment]
Maybe hooking `__sanitizer_report_error_summary` is a possible workaround?
